### ISSUE TITLE
RFC: Distribute via brew, deploy via hdiutil

### ISSUE THREAD

**Issue by marpaia** [facebook/osquery#190]
## tldr

We're in a pretty convenient situation right now on OS X, because every single one of our build dependencies exists in homebrew. I propose that we take advantage of that and make homebrew the primary way of getting the latest stable release of osquery. Then we can supply tooling to make deployment easy.
## How are we going to distribute osquery releases on OS X?

Homebrew. Now that [rocksdb is in homebrew](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/rocksdb.rb), all of our dependencies are in homebrew. We can simple create a homebrew package which "depends" on all of our dependencies. Job done.
## How are others going to deploy osquery?

The primary person who's responsible for an osquery deployment is responsible for periodically running "brew upgrade osquery". That person will also be responsible for maintaining a "deployment config file", which has a list of scheduled queries, arguments to run the daemon with, etc. We supply a tool which end users can run (think `make osx_deployment` or `./tools/package_osx.sh`) which will build an OS X .pkg file which will include all binaries and shared objects that the binaries need to link against.
## Advantages?
- The primary supported way to install osquery will be "brew install osquery"
- The packaging tooling will still work if you'd like to build from source.
- The binary will be completely dynamically linked, and the packaging tooling will account for that.
## Alternatives

The big alternative here is to spend time making the build statically linked. I don't like this because it's not the The Right Thing To Do and because it's makes the primary way of getting osquery downloading a OS X installer. This doesn't lend itself nicely to updates, so I'm afraid people will download an early version of osquery and never update it. With homebrew, you'd just have to `brew upgrade osquery`.
